cmd/zed/experiment.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+
+	"github.com/authzed/authzed-go"
+	v0 "github.com/authzed/authzed-go/arrakisapi/api"
+	"github.com/jzelinskie/stringz"
+	"github.com/open-policy-agent/opa/ast"
+	opacmd "github.com/open-policy-agent/opa/cmd"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/types"
+	"github.com/spf13/cobra"
+)
+
+func registerExperimentCmd(rootCmd *cobra.Command) {
+	rootCmd.AddCommand(experimentCmd)
+
+	experimentCmd.AddCommand(opacmd.RootCommand)
+	opacmd.RootCommand.Use = "opa"
+	opacmd.RootCommand.PersistentPreRunE = opaPreRunCmdFunc
+}
+
+var experimentCmd = &cobra.Command{
+	Use:               "experiment <subcommand>",
+	Short:             "experimental functionality",
+	PersistentPreRunE: persistentPreRunE,
+}
+
+func opaPreRunCmdFunc(cmd *cobra.Command, args []string) error {
+	if err := persistentPreRunE(cmd, args); err != nil {
+		return err
+	}
+
+	token, err := TokenFromFlags(cmd)
+	if err != nil {
+		return err
+	}
+
+	client, err := ClientFromFlags(cmd, token.Endpoint, token.Secret)
+	if err != nil {
+		return err
+	}
+
+	RegisterAuthzedBuiltins(token.System, client)
+
+	return nil
+}
+
+func RegisterAuthzedBuiltins(system string, client *authzed.Client) {
+	rego.RegisterBuiltin4(
+		&rego.Function{
+			Name:    "authzed.check",
+			Decl:    types.NewFunction(types.Args(types.S, types.S, types.S, types.S), types.B),
+			Memoize: false,
+		},
+		func(bctx rego.BuiltinContext, a, b, c, d *ast.Term) (*ast.Term, error) {
```

Comment:
maybe change `a, b, c, d`?

Reply:
This was the style used in the example, but I agree it's not very useful.